### PR TITLE
Update Rust Compiler to 1.67 and BPF to 1.15.0

### PR DESCRIPTION
**Pull request by anton-lisanin**

### CODE REVIEW

File: evm_loader/cli/src/rpc/db_call_client.rs [hunk 1]
```diff
@@ -59,19 +59,21 @@ impl CallDbClient {
 
     fn get_account_at_(&self, pubkey: &Pubkey) -> Result<Option<Account>, Error> {
         let pubkey_bytes = pubkey.to_bytes();
+        let slot: i64 = self.slot.try_into().expect("slot < i64::MAX");
```
Comment (dk-neon) [inline evm_loader/cli/src/rpc/db_call_client.rs:62]: This error message might confuse user. It's better to write something like "slot number must fit into i64".

File: evm_loader/program-macro/src/lib.rs [hunk 2]
```diff
@@ -29,7 +32,7 @@ pub fn operators_whitelist(tokens: TokenStream) -> TokenStream {
         .list
         .iter()
         .map(LitStr::value)
-        .map(|key| bs58::decode(key).into_vec().unwrap())
+        .map(|key| bs58::decode(key).into_vec().expect("Pubkey is base64 encoded"))
```
Comment (dk-neon) [inline evm_loader/program-macro/src/lib.rs:35]: Same as above. Better to write something like "Expected pubkey to be base64 encoded string".

File: evm_loader/program/src/state_account.rs [hunk 3]
```diff
@@ -164,7 +164,7 @@ impl<'a> State<'a> {
             .map(|c| c.split_at(2))
             .map(|(meta, key)|
                 BlockedAccountMeta {
-                    key: Pubkey::new(key),
+                    key: Pubkey::try_from(key).expect("key is 32 bytes"),
```
Comment (dk-neon) [inline evm_loader/program/src/state_account.rs:167]: Same as above. Consider this: "`key` is expected to be 32 bytes long public key".

File: evm_loader/cli/src/errors.rs [hunk 4]
```diff
@@ -55,43 +48,6 @@ pub enum NeonCliError {
     /// Incorrect program
     #[error("Incorrect program {0:?}")]
     IncorrectProgram(Pubkey),
-    /// Account not found at address
-    #[error("Account not found at address {0:?}.")]
```
Comment (dk-neon) [inline evm_loader/cli/src/errors.rs:59]: Is that really safe to remove these error codes from the public enum?

Comment (anton-lisanin) [reply evm_loader/cli/src/errors.rs:59]: Yes. And this enum is not public.
It's only visible inside the current crate.

File: evm_loader/program/src/state_account.rs (hunk 3 above)
Comment (anton-lisanin) [reply evm_loader/program/src/state_account.rs:167]: No one will ever see this. This `expect` will never trigger.

File: evm_loader/program-macro/src/lib.rs (hunk 2 above)
Comment (anton-lisanin) [reply evm_loader/program-macro/src/lib.rs:35]: This code is executed at compile time and the message is more than informative for a developer trying to build a project.

File: evm_loader/cli/src/rpc/db_call_client.rs (hunk 1 above)
Comment (anton-lisanin) [reply evm_loader/cli/src/rpc/db_call_client.rs]: Outdated

**Review by dk-neon — APPROVED** [review]

**Review by Deniskore — APPROVED** [review]

**Review by afalaleev — APPROVED** [review]

**Pull request merged**